src/unix.rs

```diff
@@ -33,20 +33,41 @@ fn _get(env: &impl EnvAccess) -> Option<String> {
         .filter(|val| !val.is_empty())
         .or_else(|| env.get(LANG))?;
 
-    parse_locale_code(&code)
+    Some(posix_to_bcp47(&code))
 }
 
-fn parse_locale_code(code: &str) -> Option<String> {
-    // Some locales are returned with the char encoding too, such as `en_US.UTF-8`
-    code.split_once('.')
-        .map(|(s, _)| s)
-        .or(Some(code))
-        .map(|s| s.replace('_', "-"))
+/// Converts a POSIX locale string to a BCP47 locale string.
+///
+/// This function processes the input `code` by removing any character encoding
+/// (the part after the first `.`) and any modifiers (the part after the first `@`).
+/// It replaces underscores (`_`) with hyphens (`-`) to conform to BCP47 formatting.
+///
+/// # Examples
+///
+/// ```do-not-compile
```

Comment:
From [this discussion](https://users.rust-lang.org/t/cannot-find-function-in-doc-test/52508/2): `Doc tests are compiled like external programs`.
So Rust does not see any private functions or functions from private modules when it compiles code in doc comments.

To avoid this problem, it's enough to write anything other than `rust` after the <code>```</code> characters.

Update:
I just found the [documentation for the code blocks in doc comments](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#attributes).

> The ignore attribute tells Rust to ignore your code.

So I think I should change `do-not-compile` to `ignore`.

Reply:
Got it 👍. I forgot about that limitation, so this makes more sense in hindsight. `ignore` seems like the more common way, so let's switch.

Reply:
Replaced `do-not-compile` with `ignore`.